components/server/src/jobs/fix-stripe-job.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { TypeORM } from "@gitpod/gitpod-db/lib";
+import { AttributionId } from "@gitpod/gitpod-protocol/lib/attribution";
+import { log } from "@gitpod/gitpod-protocol/lib/util/logging";
+import { inject, injectable } from "inversify";
+import { StripeService } from "../user/stripe-service";
+import { Job } from "./runner";
+import { Config } from "../config";
+import { Connection } from "typeorm";
+
+@injectable()
+export class FixStripeJob implements Job {
+    @inject(TypeORM) protected readonly typeorm: TypeORM;
+    @inject(StripeService) protected readonly stripeService: StripeService;
+    @inject(Config) protected readonly config: Config;
+
+    public readonly name = "fix-stripe-attribution-ids";
+    public frequencyMs = 300 * 60 * 1000; // every 5 hours
+
+    public async run(): Promise<void> {
+        if (!this.config.stripeSecrets) {
+            log.info("Job disabled", { name: this.name, reason: "stripeSecrets not configured" });
+            return;
+        }
+        try {
+            const c = await this.typeorm.getConnection();
+            const result = await c.query(`
+                SELECT
+                    c.stripeCustomerid, o.id as organizationId, u.id as userId
+                FROM
+                    d_b_stripe_customer c, d_b_team o, d_b_team_membership m, d_b_user u
+                WHERE
+                    o.id = m.teamId AND
+                    m.userId = u.id AND
+                    (o.name = u.name OR o.name = u.fullName) AND
+                    u._lastModified >'2023-04-15' AND
+                    c.deleted = 0 AND
+                    u.additionalData->>'$.isMigratedToTeamOnlyAttribution' = 'true' AND
+                    SUBSTRING(c.attributionId,6) = o.id
```

Comment:
Do we need a LIMIT here? How many records are affected?

Reply:
1422 we can run them in one batch